dev/source/docs/debugging-with-gdb-using-vscode.rst

```diff
@@ -0,0 +1,223 @@
+.. _debugging-with-gdb-using-vscode:
+
+===============================
+Debugging with GDB using VSCode
+===============================
+
+This page describes how to setup and use GDB with the VSCode IDE to debug issues with SITL, autotests, and unit tests.
+
+Introduction
+============
+
+This guide assumes that you have already successfully use SITL
+on your machine following the instructions :ref:`to setup SITL <setting-up-sitl-on-linux>`.
+
+Installing GDB
+==============
+
+To install GDB on Linux, please use package installer. If you are on a debian based system (such as Ubuntu or Mint),
+simply use the ``apt install`` command :
+
+::
+
+    sudo apt install gdb
+
+Setting up SITL
+===============
+
+In order to use GDB, you need to configure your SITL build with debug symbols. Use :
+
+::
+
+    ./waf configure --debug
+
+Setting up VSCode Debugger
+==========================
+With the VSCode open to the ArduPilot directory. Open the `Run and Debug` menu (Ctrl+Shift+D). In the Run and Debug menu, `create a launch.json file`. Then `Add Configuration`, it should open a launch.json file for setting up debugging configurations.
+
+.. image:: ../images/VSCode-config1.png
+    :target: ../_images/VSCode-config1.png
+
+
+
+
+Example launch.json File
+------------------------
+
+.. code-block:: json
```

Comment:
just use ::     sphinx does not lex json as a code block and gives build error...otherwise its fine...

Reply:
I've changed the code block to javascript  as that will at least parse nearly the same. It seems newer versions of sphinx can parse json. But not important enough to go down that hole...